### PR TITLE
youtube-dl: use as default Python3 and update to 2019.4.7

### PR DESCRIPTION
**Pull request by BKPepe**
Maintainer: @ianchi
Co-Maintainer: me (@BKPepe)

Compiled tested: cortexa53, Turris MOX, OpenWrt master
Run tested: cortexa53, Turris MOX, OpenWrt master

Description:

- Update to version 2019.4.7 ([release notes](https://github.com/ytdl-org/youtube-dl/releases/tag/2019.04.07))
- download tarball from files.pythonhosted.org instead of codeload.github.com to be more consistent with other Python packages. 
- Use Py3Package
- some SDKs for some targets complained
`WARNING: Makefile 'package/feeds/packages/youtube-dl/Makefile' has a build dependency on 'zip/host', which does not exist`
This is now fixed in this PR.
- changed their website from https://youtube-dl.org to https://yt-dl.org/ and together with this change, they moved yt-dl.org to separate Github repository from github.com/rg3. More details: https://github.com/ytdl-org/youtube-dl/commit/067aa17edf5a46a8cbc4d6b90864eddf051fa2bc
- Removed patch, which I did in the past.

### CODE REVIEW

**Comment by diizzyy** [comment]
The common approach seems to stick with one name/variant. I personally think we should try moving as much as possible to python3 as you willl most likely see unnecessary breakage trying to migrate everythng at once to python3 (as seen with OpenSSL despite *much work** being done to prevent that).

**Comment by BKPepe** [comment]
Thank you for your suggestion. Let's do it as you said. I'm gonna remove Python2 variant. I just wanted to have in upcoming OpenWrt release both versions as it makes sense for me. Not everyone wants to be to forced to Python3 even the support for Python2 should no longer be provided soon.

File: multimedia/youtube-dl/Makefile [hunk 1]
```diff
@@ -8,46 +8,48 @@
...
-define Package/youtube-dl
+PKG_UNPACK:=$(HOST_TAR) -C $(PKG_BUILD_DIR) --strip-components=1 -xzf $(DL_DIR)/$(PKG_SOURCE)
+
+define Package/youtube-dl/Default
   SECTION:=multimedia
   CATEGORY:=Multimedia
   TITLE:=utility to download videos from YouTube.com
-  DEPENDS:=+python-openssl +python-email +python-xml +python-codecs +python-ctypes +ca-certificates
-  URL:=https://youtube-dl.org
+  DEPENDS:=+ca-certificates
```
Comment (neheb) [inline multimedia/youtube-dl/Makefile]: Can ca-bundle be used instead?

Comment (BKPepe) [reply multimedia/youtube-dl/Makefile]: Hm... In theory, it should be possible, but In Ubuntu and Debian, there isn't mention anything about ca-bundle, but only about ca-certificates. I will make a note about it and check it when they release a new version.

**Pull request merged**